core/src/test/java/org/hibernate/ogm/backendtck/queries/projection/Poem.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.queries.projection;
+
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.search.annotations.Analyze;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.annotations.SortableField;
+import org.hibernate.search.annotations.Store;
+
+@Entity
+@Table(name = Poem.TABLE_NAME)
+@Indexed
+public class Poem {
+
+	public static final String TABLE_NAME = "POEM";
+
+	@Id
+	private Long id;
+
+	@Field(analyze = Analyze.NO, store = Store.YES)
+	@SortableField
+	private String name;
+
+	@Field(analyze = Analyze.NO)
+	@SortableField
+	private String author;
+
+	@Field(analyze = Analyze.NO)
+	@SortableField
+	private Integer year;
+
+	public Poem() {
+	}
+
+	public Poem(Long id, String name, String author, int year) {
+		this.id = id;
+		this.name = name;
+		this.author = author;
+		this.year = year;
```

Comment:
I prefer if we don't have fields that aren't used.
I might be wrong, but it seems you are only interested to `id` and `name` for the tests

Reply:
Other fields are present to remark the concept of partial Entity. I'd rather leave them.

Reply:
Make sense.

I wonder if we can find a way to make it a bit less obscure. Somebody might delete them in the future.

Maybe you can add a test to project only this two fields? So we have a test case that doesn't return the id. In the end the id is sort of a special field.

Reply:
I do not know because actually all tests without projection work on equals and hashcode, that in turn use all fields.

Reply:
I can understand that **technicallly** they are used, But I could remove them in the Poem class and I wouldn't have any problem with our test suit. If somebody looks at this code how can he guess that this fields are actually part of the test suit?

At least leave a comment.

Reply:
ok I'm making sure that some tests use some, other tests other fields, so you do not have fields not directly used